phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,221 @@
+        {
+            $('#loading_indicator').css('display', 'none');
+            reset_polling();
+        }
+    }
+
+    function parse_message(messageJSON)
+    {
+        $('#loading_indicator').css('display', 'none');
+
+        messageJSON = messageJSON.trim();
+        var responseObject = JSON.parse(messageJSON);
+
+        // Parse object
+        if (responseObject.hasOwnProperty('errors'))
+        {
+            add_message('error', responseObject.errors)
+        }
+
+        if (responseObject.hasOwnProperty('warnings'))
+        {
+            add_message('warning', responseObject.warnings)
+        }
+
+        if (responseObject.hasOwnProperty('logs'))
+        {
+            add_message('log', responseObject.logs);
+        }
+
+        if (responseObject.hasOwnProperty('form'))
+        {
+            add_form(responseObject.form);
+        }
+    }
+
+    function add_message(type, messages)
+    {
+        // Get message containers
+        var errorContainer = $('#error-container');
+        var warningContainer = $('#warning-container');
+        var logContainer = $('#log-container');
+
+        var title, description, msgElement, arraySize = messages.length;
+        for (var i = 0; i < arraySize; i++)
+        {
+            msgElement = $(document.createElement('div'));
+            title = $(document.createElement('strong'));
+            title.text(messages[i].title);
+            msgElement.append(title);
+
+            if (messages[i].hasOwnProperty('description'))
+            {
+                description = $(document.createElement('p'));
+                description.text(messages[i].description);
+                msgElement.append(description);
+            }
+
+            switch (type)
+            {
+                case 'error':
+                    msgElement.addClass('errorbox');
+                    errorContainer.append(msgElement);
+                break;
+                case 'warning':
+                    msgElement.addClass('warningbox');
+                    warningContainer.append(msgElement);
+                break;
+                case 'log':
+                    msgElement.addClass('log');
+                    logContainer.append(msgElement);
+                break;
+            }
+        }
+    }
+
+    function add_form(formHtml)
+    {
+        var formContainer = $('#content-container');
+        formContainer.html(formHtml);
+        var form = $('#install_install');
+        intercept_form_submit(form);
+    }
+
+    function start_polling(xhReq)
+    {
+        reset_polling();
+        pollTimer = setInterval(function() { poll_content(xhReq); }, 500)
+    }
+
+    function reset_polling()
+    {
+        clearInterval(pollTimer);
+        nextReadPosition = 0;
+    }
+
+    function submit_form(form, submitBtn)
+    {
+        form.css('display', 'none');
+
+        var xhReq = create_xhr_object();
+        xhReq.open("POST", form.attr('action'), true);
```

Comment:
single quotes. but this entire function can be reduces to like two lines of jquery